specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-05-01-preview/machineLearningServices.json

```diff
@@ -0,0 +1,5214 @@
+          "type": "string"
+        },
+        "userManagedDependencies": {
+          "description": "True means that AzureML reuses an existing python environment; False means that AzureML will create a python environment based on the Conda dependencies specification.",
+          "type": "boolean"
+        },
+        "condaDependencies": {},
+        "baseCondaEnvironment": {
+          "type": "string"
+        }
+      }
+    },
+    "ContainerRegistry": {
+      "type": "object",
+      "properties": {
+        "address": {
+          "type": "string"
+        },
+        "username": {
+          "type": "string"
+        },
+        "password": {
+          "type": "string"
+        }
+      }
+    },
+    "ModelDockerSection": {
+      "type": "object",
+      "properties": {
+        "baseImage": {
+          "description": "Base image used for Docker-based runs. Mutually exclusive with BaseDockerfile.",
+          "type": "string",
+          "example": "ubuntu:latest"
+        },
+        "baseDockerfile": {
+          "description": "Base Dockerfile used for Docker-based runs. Mutually exclusive with BaseImage.",
+          "type": "string",
+          "example": "FROM ubuntu:latest\r\nRUN echo \"Hello world!\""
+        },
+        "baseImageRegistry": {
+          "description": "Image registry that contains the base image.",
+          "allOf": [
+            {
+              "$ref": "#/definitions/ContainerRegistry"
+            }
+          ]
+        }
+      }
+    },
+    "SparkMavenPackage": {
+      "type": "object",
+      "properties": {
+        "group": {
+          "type": "string"
+        },
+        "artifact": {
+          "type": "string"
+        },
+        "version": {
+          "type": "string"
+        }
+      }
+    },
+    "ModelSparkSection": {
+      "type": "object",
+      "properties": {
+        "repositories": {
+          "description": "The list of spark repositories.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "packages": {
+          "description": "The Spark packages to use.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SparkMavenPackage"
+          }
+        },
+        "precachePackages": {
+          "description": "Whether to precache the packages.",
+          "type": "boolean"
+        }
+      }
+    },
+    "RCranPackage": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "The package name.",
+          "type": "string"
+        },
+        "repository": {
+          "description": "The repository name.",
+          "type": "string"
+        }
+      }
+    },
+    "RGitHubPackage": {
+      "type": "object",
+      "properties": {
+        "repository": {
+          "description": "Repository address in the format username/repo[/subdir][@ref|#pull].",
+          "type": "string"
+        },
+        "authToken": {
```

Comment:
please add "x-ms-secret": true to all values that are secrets

Reply:
added